src/cli/setup/federation_change.rs

```diff
@@ -0,0 +1,524 @@
+use crate::cli::setup::traits::Response;
+
+use crate::errors::Error;
+
+use clap::{App, Arg, ArgGroup, ArgMatches, SubCommand};
+use std::fs;
+use std::fs::File;
+use std::fs::OpenOptions;
+use std::io::{BufRead, BufReader, Write};
+use std::path::{Path, PathBuf};
+use std::str::FromStr;
+use std::sync::Mutex;
+use tapyrus::blockdata::block::XField;
+
+//federation command is used to register/unregister federation change.
+// When register is invoked it writes a new file with the given parameters.
+// When unregister is invoked it removes the file with the given parameters.
+//
+// FILE_NAME: federationchange_<timestamp>.dat"
+// PATH: FEDERATION_CHANGE_FILE
+//
+//USAGE:
+//federation [OPTIONS] <change> --height <height> <--aggregated-public-key <aggregated-public-key>|--max-block-size <max-block-size>>
+//
+//FLAGS:
+//    -h, --help       Prints help information
+//    -V, --version    Prints version information
+//
+//OPTIONS:
+//    --aggregated-public-key <aggregated-public-key>    aggregated public key of the new federation
```

Comment:
If you update the aggregated public key with `--aggregated-public-key`, don't you also need the `threshold` and `node-vss` described in federations.toml?

Reply:
oh! this needs further discussion. I'll describe in the [issue]( https://github.com/chaintope/tapyrus-signer/issues/163#issuecomment-1761307920)